presto-main/src/main/java/com/facebook/presto/server/KerberosInternalCommunicationConfig.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.server;
+
+import io.airlift.configuration.Config;
+
+import javax.validation.constraints.NotNull;
+
+import java.io.File;
+
+public class KerberosInternalCommunicationConfig
+{
+    private String kerberosPrincipal;
```

Comment:
You can build it based on the service name. `kerberosPrincipal=<kerberosServiceName>/_HOST`, and it should get the realm from the config. See https://github.com/prestodb/presto/blob/master/presto-main/src/main/java/com/facebook/presto/server/security/KerberosAuthenticator.java#L68

We are using this convention to build the service principal for the server. I thing we can safely follow the approach to build a client principal as well.